src/reducers/fileTree.tsx

```diff
@@ -191,6 +190,100 @@ export const getRelativePath = ({
   return pathList[newIndex];
 };
 
+const findRelativeMessageUid = (
+  currentPath: string,
+  messageMap: LinterMessageMap,
+  pathList: string[],
+  position: RelativePathPosition,
+): LinterMessage['uid'] | null => {
+  let path = currentPath;
+  for (let i = 0; i < pathList.length; i++) {
+    const nextPath = getRelativePath({
+      currentPath: path,
+      pathList,
+      position,
+    });
+    const nextMessages = messageMap[nextPath];
+    if (nextMessages) {
+      const nextMessagesForPath = getMessagesForPath(nextMessages);
+      if (position === RelativePathPosition.previous) {
+        const lastMessageIndex = nextMessagesForPath.length - 1;
+        if (lastMessageIndex >= 0) {
+          return nextMessagesForPath[lastMessageIndex].uid;
+        }
+      } else if (nextMessagesForPath.length) {
+        return nextMessagesForPath[0].uid;
+      }
+    }
+    path = nextPath;
+  }
+  // If we got here then we never found another message, which would be odd,
+  // because we should have at the very least gotten back to our original
+  // message.
+  // We should probably log this.
+  return null;
+};
+
+export type GetRelativeMessageUidParams = {
+  currentMessageUid: LinterMessage['uid'] | void;
+  currentPath: string;
+  locateCurrentMessage?: boolean;
+  messageMap: LinterMessageMap;
+  pathList: string[];
+  position: RelativePathPosition;
+};
+
+export const getRelativeMessageUid = ({
+  currentMessageUid,
+  currentPath,
+  messageMap,
+  pathList,
+  position,
+}: GetRelativeMessageUidParams): string | null => {
+  if (!currentMessageUid) {
+    // There is no current message, so get the first one.
+    if (!Object.keys(messageMap).length) {
+      return null;
+    }
+    for (const path of pathList) {
+      const messages = messageMap[path];
+      if (messages) {
+        const messagesForPath = getMessagesForPath(messages);
+        if (messagesForPath.length) {
+          return messagesForPath[0].uid;
+        }
+      }
+    }
+    // No messages found in the map (which shouldn't be possible because of the
+    // length check, so throw an error.
+    throw new Error(
```

Comment:
I don't know how to test this, because it is an impossibility. Maybe that means we don't need to throw here at all? Maybe we should just return `null`?

Reply:
This code is outdated now so I think this comment is now irrelevant.